### PR TITLE
(dropbox) Don't kill dropbox process, if it's running before install

### LINKED ISSUE #1242: (dropbox) does not autostart after upgrade
## Expected Behavior
`choco upgrade dropbox`
choco gets an upgrade package (eg. in my case v68.4.102 ➡ v69.3.90), installs it and then starts dropbox again, so it continues watching and syncing files in the background

## Current Behavior
choco gets an upgrade package, installs it and then does not start dropbox

## Possible Solution
if dropbox was running before update, add a command to autostart it again after the upgrade

## Steps to Reproduce (for bugs)
n/a

## Context
I have set `choco upgrade all` to run during system start, to always have fresh versions. I have to take care to start dropbox, after it gets upgraded by choco.

## Your Environment

* Package Version used: irrelevant but v68.4.102 ➡ v69.3.90
* Operating System and version: Windows 10 Pro version 1809 build 17763.379
* Chocolatey version: 0.10.11

Just in case I do something wrong, here's a printout of the update from console:
```cmd
C:\Users\xxxxxx>choco upgrade dropbox
Chocolatey v0.10.11
Upgrading the following packages:
dropbox
By upgrading you accept licenses for the packages.

You have dropbox v68.4.102 installed. Version 69.3.90 is available based on your source(s).
Progress: Downloading dropbox 69.3.90... 100%

dropbox v69.3.90 [Approved]
dropbox package files upgrade completed. Performing other installation steps.
Downloading dropbox
  from 'https://clientupdates.dropboxstatic.com/dbx-releng/client/Dropbox%2069.3.90%20Offline%20Installer.exe'
Progress: 100% - Completed download of C:\Users\jandu\AppData\Local\Temp\chocolatey\dropbox\69.3.90\Dropbox 69.3.90 Offline Installer.exe (97.89 MB).
Download of Dropbox 69.3.90 Offline Installer.exe (97.89 MB) completed.
Hashes match.
Installing dropbox...
dropbox has been installed.
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
 The upgrade of dropbox was successful.
  Software installed as 'exe', install location is likely default.
```

### PR DESCRIPTION
**Pull request by jack1142**
## Description
Before this, dropbox was being killed after the installation whether it was running before installation or not. Now installation script checks if dropbox is already running before start of the installation and if it is, then it will leave the dropbox's process alive after the installation (as installer always starts dropbox after it ends its work).
I also did some small code clean up - I removed helper function for retrieving registry key, as we now can just use `Get-UninstallRegistryKey`.

This fixes #1242

// on a side note, I know that RedBaron2 was also working on this in #1246, but I decided to send this PR anyway, because he has a different approach to this problem - he fully removed killing process part, which in my opinion isn't good, because it will make the (post-)installation not silent. 

## Motivation and Context
Before this PR, user could have been left with their data unsynced, if they're for example doing chocolatey upgrades in background - they may not notice that dropbox's process got killed.

## How Has this Been Tested?
I used chocolatey-test-environment, I checked, if it can be installed and uninstalled and checked if dropbox will stay alive if it is running before installation (I also checked it during upgrade, as this is the main reason of this change)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:

- [x] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/CONTRIBUTING.md)
- [x] The added/modified package passed install/uninstall in the chocolatey test environment.
- [x] The changes only affect a single package (not including meta package).

### CODE REVIEW

**Comment by RedBaron2** [comment]
@jack1142 
Thanks for picking this up. I seem to have RL issues that are distracting me from doing this properly. Thanks 😃

**Comment by jack1142** [comment]
@RedBaron2 I don't use Dropbox myself (OneDrive user here), but I can add a check if newer or same version isn't already installed.

**Comment by jack1142** [comment]
Actually I will only check for equality, as it would prevent installing older versions (with `--version` parameter) on top of newer otherwise.

File: automatic/dropbox/tools/chocolateyinstall.ps1 [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
+[array]$key = Get-UninstallRegistryKey -SoftwareName "Dropbox"
+$alreadyInstalled = $false
+$version = ''
+
+if ($key.Count -ne 0) {
+  $key | ForEach-Object {
+    if ($_.DisplayVersion -eq $version) {
+      $alreadyInstalled = $true
+    }
+  }
```
Comment (AdmiringWorm) [inline automatic/dropbox/tools/chocolateyinstall.ps1]: Please move the this code block to a helper.ps1 script, and call it that way.

File: automatic/dropbox/tools/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -1,18 +1,37 @@
-﻿$ErrorActionPreference  = 'Stop'
+$ErrorActionPreference  = 'Stop'
```
Comment (AdmiringWorm) [inline automatic/dropbox/tools/chocolateyinstall.ps1]: If you removed BOM here, please add it back.
Powershell incorrectly reads script files as ANSI when no BOM is available.

File: automatic/dropbox/tools/chocolateyuninstall.ps1 [hunk 3]
```diff
@@ -1,10 +1,23 @@
-. (Join-Path (Split-Path $MyInvocation.MyCommand.Definition -Parent) 'helpers.ps1')
+$ErrorActionPreference = 'Stop'
```
Comment (AdmiringWorm) [inline automatic/dropbox/tools/chocolateyuninstall.ps1]: If you removed BOM here, please add it back.
Powershell incorrectly reads script files as ANSI when no BOM is available.

**Comment by AdmiringWorm** [comment]
Would be great if you could rebase the PR as well.

**Comment by jack1142** [comment]
@AdmiringWorm I'll fix those things today, but did you mean resolving conflicts by rebasing?

**Comment by AdmiringWorm** [comment]
> I'll fix those things today, but did you mean resolving conflicts by rebasing?

yes, the conflicts need to be resolved, and rebasing is the preferred way to do this as it keeps the repo history as clean as possible.

**Comment by jack1142** [comment]
@AdmiringWorm I made the changes you asked for

**Pull request merged**